JumpscaleLibs/sal/chatflow/Chatflow.py

```diff
@@ -0,0 +1,153 @@
+import netaddr
+from Jumpscale import j
+import random
+
+
+class Chatflow(j.baseclasses.object):
+    __jslocation__ = "j.sal.chatflow"
+
+    def _init(self, **kwargs):
+        pass
+
+    def get_all_ips(self, ip_range):
+        """
+        ip_range: (String)ip range of network
+        return all available ips of this network
+        """
+        networks = netaddr.IPNetwork(ip_range)
+        ips = []
+        for ip in list(networks.iter_hosts()):
+            ips.append(ip.format())
+        return ips
+
+    def nodes_get(self, number_of_nodes):
+        nodes = j.sal.zosv2.nodes_finder.nodes_search()
+        nodes_selected = []
+        for i in range(number_of_nodes):
+            node = random.choice(nodes)
+            nodes_selected.append(node)
+            # TODO remove the random node selected
+
+        return nodes_selected
+
+    def network_configure(self, bot, reservation, nodes, customer_tid, ip_version):
+        """
+        bot: Gedis chatbot object from chatflow
+        reservation: reservation object from schema
+        node: list of node objects from explorer
+
+        return reservation (Object) , config of network (dict)
+        """
+        ip_range = self.ip_range_get(bot)
+        return self.network_get(bot, reservation, ip_range, nodes, customer_tid, ip_version)
+
+    def ip_range_get(self, bot):
+        """
+        bot: Gedis chatbot object from chatflow
+        return ip_range from user or generated one
+        """
+        ip_range_choose = ["Specify IP Range", "Choose IP Range for me"]
```

Comment:
"to have access to the container, the network must be configured"

and the choices are:

"Configure IP range myself"

"Choose IP range for me"